src/Http/Http.Extensions/src/DefaultProblemDetailsWriter.cs

```diff
@@ -15,11 +13,22 @@ internal sealed partial class DefaultProblemDetailsWriter : IProblemDetailsWrite
 {
     private static readonly MediaTypeHeaderValue _jsonMediaType = new("application/json");
     private static readonly MediaTypeHeaderValue _problemDetailsJsonMediaType = new("application/problem+json");
+
     private readonly ProblemDetailsOptions _options;
+    private readonly JsonSerializerOptions _serializerOptions;
 
-    public DefaultProblemDetailsWriter(IOptions<ProblemDetailsOptions> options)
+    public DefaultProblemDetailsWriter(IOptions<ProblemDetailsOptions> options, IOptions<JsonOptions> jsonOptions)
     {
         _options = options.Value;
+        _serializerOptions = new JsonSerializerOptions(jsonOptions.Value.SerializerOptions);
+
+        if (_serializerOptions.TypeInfoResolver is not null)
```

Comment:
What happens if TypeInfoResolver is null?

Reply:
In `non-aot scenario` it will use the reflection-based resolver, so, we don't really need to include our internal context. In `AOT` scenario, unless your application only serializes `Problem Details`, it should not be null and if it is null not adding our internal context seems might be bad but probably it will fail elsewhere first. Make sense?

Reply:
But it seems to me that if `TypeInfoResolver` is null then this type can't work. Should there be a check that if IsDynamicCodeSupported is false and `TypeInfoResolver` is null then error?

I think a comment here would be useful for future devs.

Reply:
Interesting idea. Do you think error is enough or should we set our context in this case? I believe setting our context could hide the configuration issue and make it harder to understand.

Reply:
I don't know. If the problem is configuration, then IMO throwing here with a helpful error message is better than pretending things are ok and then later failing when serialization happens with an obscure error message.

Reply:
@JamesNK  I have been thinking about this and maybe we shouldn't throw during the configuration because it could be too earlier to make sure a problem details will not be able to be serializer. I am saying that because in a situation where the user added a custom `IProblemDetailsWriter` that could serializer a `ProblemDetails` in a different format (eg, `XML`) throwing because the `JsonContext` was not configured correctly might be wrong. What do you think?

Reply:
I agree with @brunolins16, I don't think we should throw because the TypeInfoResolver is null. That shouldn't come from this code. Feels like something more top level and central as a whole. Its part of the "how does the user know they need to set a context" in the first place.